frontend/controller/actions/chatroom.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+import sbp from '~/shared/sbp.js'
+import L from '@view-utils/translations.js'
+import { GIErrorUIRuntimeError } from '@model/errors.js'
+import { encryptedAction } from './utils.js'
+import type { GIActionParams } from './types.js'
+
+export default (sbp('sbp/selectors/register', {
+  'gi.actions/chatroom/create': async function (params: GIActionParams) {
+    try {
+      const message = await sbp('chelonia/out/registerContract', {
+        contractName: 'gi.contracts/chatroom',
+        data: params.data
+      })
+
+      return message
+    } catch (e) {
+      console.error('gi.actions/chatroom/register failed!', e)
+      throw new GIErrorUIRuntimeError(L('Failed to create chat channel.'))
+    }
+  },
+  ...encryptedAction('gi.actions/chatroom/addMessage', L('Failed to add message.')),
+  ...encryptedAction('gi.actions/chatroom/join', L('Failed to join chat channel.')),
+  ...encryptedAction('gi.actions/chatroom/rename', L('Failed to rename chat channel.')),
+  ...encryptedAction('gi.actions/chatroom/changeDescription', L('Failed to change chat channel description.')),
+  ...encryptedAction('gi.actions/chatroom/leave', L('Failed to leave chat channel.')),
+  ...encryptedAction('gi.actions/chatroom/delete', L('Failed to delete chat channel.'))
```

Comment:
We use the name `chatroom` in the codebase, but users are more friendly with `channel`.

Reply:
@Silver-IT personally I agree with @snowteamer. I think it might lead to confusion when we use two different words. For example, elsewhere in your code you had a user-facing error that said, `'Failed in joining General chatroom.`

I think both are friendly to users, and it would make sense to be consistent in both the code and the user-facing messages in how we refer to these, and I think `chatroom` is fine.